scripts/quests/sandoria/PLD_AF3_Under_Oath.lua

```diff
@@ -0,0 +1,309 @@
+                    quest:setVar(player, 'Prog', 3)
+                end,
+
+                [41] = function(player, csid, option, npc)
+                    if option == 1 then
+                        npcUtil.giveKeyItem(player, xi.ki.STRANGE_SHEET_OF_PAPER)
+                    end
+                end,
+
+                [77] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 5)
+                end,
+            },
+        },
+
+        [xi.zone.DAVOI] =
+        {
+            ['Village_Well'] =
+            {
+                onTrade = function(player, npc, trade)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if
+                        npcUtil.tradeHasExactly(trade, xi.items.WELL_WEIGHT) and
+                        questProgress == 5
+                    then
+                        return quest:progressEvent(113)
+                    end
+                end,
+
+                onTrigger = function(player, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if
+                        questProgress == 5 and
+                        player:hasKeyItem(xi.ki.STRANGE_SHEET_OF_PAPER) and
+                        not player:hasItem(xi.items.WELL_WEIGHT)
+                    then
+                        npcUtil.popFromQM(player, npc, spawnedMobs, { claim = true, hide = 0 })
+                    elseif
+                        questProgress == 6 and
+                        player:hasKeyItem(xi.ki.KNIGHTS_CONFESSION)
+                    then
+                        return quest:event(112)
+                    end
+                end,
+
+            },
+
+            onEventFinish =
+            {
+                [113] = function(player, csid, option, npc)
+                        player:confirmTrade()
+                        npcUtil.giveKeyItem(player, xi.ki.KNIGHTS_CONFESSION)
+                        quest:setVar(player, 'Prog', 6)
+                        player:delKeyItem(xi.ki.STRANGE_SHEET_OF_PAPER)
+                end,
+            },
+        },
+
+        [xi.zone.NORTHERN_SAN_DORIA] =
+        {
+            ['Ailbeche'] =
+            {
+                onTrigger = function(player, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if
+                        questProgress == 6 and
+                        player:hasKeyItem(xi.ki.KNIGHTS_CONFESSION)
+                    then
+                        return quest:progressEvent(59) -- Starts Quest.
+                    elseif questProgress == 8 then
+                        return quest:event(13)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [59] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 7)
+                end,
+            },
+        },
+    },
+
```

Comment:
This sections is terribly awkward for readability. 
What we can do to aid this is to break it up into sections, such as:
 check = function(player, status, vars)
            return status == QUEST_AVAILABLE and
                quest.Prog == 1
        end,
etc.

Reply:
what?

Reply:
Sorry I explained this terrible.

To organize this quest, we can break it down into segments based on progress. 
Because we have multiple things happening when prog is at any specific point, we can break QUEST_ACCEPTED into separate sections based on prog.

So when we check for QUEST_ACCEPTED, we can also check for prog. We'll break it into sections of:

QUEST_ACCEPTED and quest.Prog == 1
In this section we have all dialogue with NPCs when it requires prog to be 1

QUEST_ACCEPTED and quest.Prog == 2
All dialogue with NPCs or requirements of prog == 2

QUEST_ACCEPTED and quest.Prog == 3
So on an so forth.

This helps break the quest down into concise and clear sections of the quest. I hope that makes sense. If not, feel free to DM me and we can hop in a call and I'll show some more examples.